### PR TITLE
#165413073 add article tag

### PR DESCRIPTION
**Pull request by jabichris**
#### Description
This PR is for adding functionalities such as:
- Adding tag to an article.
- View all tags of an article.
#### Type of change
N/A

#### How has it been tested
By going to `http://localhost:3002/articles/update/:slug `and edit tag input field for creating a tag for an existing article. Then `http://localhost:3002/articles/:slug` for one article then go at at bottom of the page to view all tags.

#### Pivotal tracker story ID
#165413073

#### Screenshots
image
image

### CODE REVIEW

**Review by ericnyirimana — COMMENTED** [review]
@jabichris Rebase your PR

**Review by Kabalisa — COMMENTED** [review]
@jabichris. I am having trouble testing your PR on deployment. when trying to create a new article together with its tag, toasts indicating errors are shown after clicking publish. please address the issue.

**Review by Kabalisa — APPROVED** [review]

**Review by ericnyirimana — APPROVED** [review]

File: src/components/articles/createOrUpdateArticle.jsx [hunk 1]
```diff
@@ -59,10 +63,14 @@ class CreateOrUpdateArticle extends Component {
     const slug = this.props.match.params.slug;
     if (slug) {
       if (nextProps.article.fetched === 'done') {
-        const { title, tag, category,description, body } = nextProps.article.article;
+        const {
+          title,
+          category,
+          description,
+          body,
+        } = nextProps.article.article || nextProps.article.data;
```
Comment (macphilips) [inline src/components/articles/createOrUpdateArticle.jsx]: Why do you have articles.data and article.article? please unify both

File: src/assets/scss/_articleCard.scss [hunk 2]
```diff
@@ -18,6 +18,7 @@
   div {
     cursor: pointer;
     padding: 12px;
+    -webkit-column-break-inside: avoid;
```
Comment (macphilips) [inline src/assets/scss/_articleCard.scss:21]: using `-webkit-*` means you are targeting a chrome browser. Have you tried testing on other browsers to see if the column break inside is avoided? besides what does this attribute do?

Comment (jabichris) [reply src/assets/scss/_articleCard.scss:21]: This was a suggestion from our P.O. One article card was breaking into pieces which wasn't good.

Comment (jabichris) [reply src/assets/scss/_articleCard.scss:21]: It avoid the break inside the article card at the home page. I tried it with Safari and everything is working well.

**Review by Kabalisa — CHANGES_REQUESTED** [review]
@jabichris. I have tested this PR on Heroku and there is an issue about the CSS of the tags when someone creates many tags. You can address that issue.

**Review by ericnyirimana — CHANGES_REQUESTED** [review]
@jabichris I have tested the deployment but there is an issue with the CSS style for tags. Check again

**Comment by jabichris** [comment]
@

**Comment by jabichris** [comment]
@ericnyirimana and @Kabalisa , the issue is already fixed it.

**Review by ericnyirimana — APPROVED** [review]

**Review by Kabalisa — APPROVED** [review]

**Comment by macphilips** [comment]
@jabichris resolve conflict

**Comment by jabichris** [comment]
> @jabichris resolve conflict

@macphilips, already resolved

**Pull request merged**